Refactoring and Code Cleanup

alexbestul opened the pull request:
This PR contains various changes intended to improve the readability and usability of the scripts.

In this PR:
* createPackage.ps1 accepts arguments via the command line in addition to as environment variables
* createPackage.ps1 includes comment-based help
* The code is simplified, and takes more advantage of PowerShell's features

I tried to keep the individual changes as distinct commits, so you can pick and choose which (if any) you want.

Please let me know if you have any questions, or if there is anything you would like me to change.

alexbestul committed:
Follow PowerShell idioms for parameters

This change maintains the existing usage pattern (as described in the
README). It adds basic parameter validation, and also allows values to
be passed in as command-line parameters if desired.

alexbestul committed:
Improve path handling

alexbestul committed:
Simplify checksum fetching code

alexbestul committed:
Follow PowerShell idioms for string building

alexbestul committed:
Refactor for cleanliness

alexbestul committed:
Add docstrings

alexbestul committed:
Add new syntax options to README

varshavaradarajan commented on createPackage.ps1:
```diff
@@ -1,13 +1,57 @@
+
+Push-Location $path
+
+$checksum32 = (Invoke-RestMethod "https://download.gocd.io/binaries/$fullVersion/win/go-$type-$fullVersion-jre-32bit-setup.exe.sha256sum").Split(' ')[0]
```
Hey, the reason I chose `System.Net.WebClient` was that we are using powershell 2.0 to build these packages for pushing. `Invoke-RestMethod` won't be available. I don't see any reason to update the version of powershell on the machines we used to build the packages because we are not losing out of any features for building the packages that poweshell v3 offers and not 2. Can you please change this back to using the WebClient?

alexbestul committed:
Use WebClient instead of Invoke-RestMethod

To maintain PowerShell 2.0 compatibility.

alexbestul committed:
Specify target PowerShell version

This marker indicates that the script needs *at least* PowerShell v2.0
to run. It does not, however, limit the PowerShell API to a v2.0 subset
if running on a newer version of PowerShell. It is still up to
developers to ensure that they are writing 2.0-compatible code.

alexbestul commented:
Varsha,

Sure, that's a fine reason to use WebClient--I made the change as you requested. I also added the statement `#Requires -Version 2.0` to the file, to indicate that you're targeting PowerShell v2.0. That won't prevent people from writing non-2.0-compliant code, but it should at least give a hint that you're targeting 2.0.

I tested the scripts in a Windows 7 VM to verify PowerShell 2.0 compatibility, and all looks good to me.

Would you like me to rebase my commits to remove any mention of Invoke-RestMethod from the branch's history? If so, I can do that and open a new PR.

varshavaradarajan commented:
Can you please rebase and squash it to 3 commits? LGTM 👍

alexbestul commented:
I have squashed into 3 commits, and opened PR #4 with the revised version.
